test/e2e/storage/testsuites/stress.go

```diff
@@ -0,0 +1,192 @@
+
+	intreeOps   opCounts
+	migratedOps opCounts
+
+	resources []*VolumeResource
+	pods      []*v1.Pod
+	// stop and wait for any async routines
+	wg      sync.WaitGroup
+	stopChs []chan struct{}
+}
+
+var _ TestSuite = &stressTestSuite{}
+
+// InitStressTestSuite returns stressTestSuite that implements TestSuite interface
+func InitStressTestSuite() TestSuite {
+	return &stressTestSuite{
+		tsInfo: TestSuiteInfo{
+			Name:       "stress",
+			FeatureTag: "[Feature: VolumeStress]",
+			TestPatterns: []testpatterns.TestPattern{
+				testpatterns.DefaultFsDynamicPV,
+				testpatterns.BlockVolModeDynamicPV,
+			},
+		},
+	}
+}
+
+func (t *stressTestSuite) GetTestSuiteInfo() TestSuiteInfo {
+	return t.tsInfo
+}
+
+func (t *stressTestSuite) SkipRedundantSuite(driver TestDriver, pattern testpatterns.TestPattern) {
+}
+
+func (t *stressTestSuite) DefineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	var (
+		dInfo = driver.GetDriverInfo()
+		cs    clientset.Interface
+	)
+
+	ginkgo.BeforeEach(func() {
+		// Check preconditions.
+		ok := false
+		_, ok = driver.(DynamicPVTestDriver)
+		if !ok {
+			e2eskipper.Skipf("Driver %s doesn't support %v -- skipping", dInfo.Name, pattern.VolType)
+		}
+
+		if !driver.GetDriverInfo().Capabilities[CapBlock] && pattern.VolMode == v1.PersistentVolumeBlock {
+			e2eskipper.Skipf("Driver %q does not support block volume mode - skipping", driver.GetDriverInfo().Name)
+		}
+	})
+
+	// This intentionally comes after checking the preconditions because it
+	// registers its own BeforeEach which creates the namespace. Beware that it
+	// also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewDefaultFramework("stress")
+
+	init := func() stressTest {
+		cs = f.ClientSet
+		l := stressTest{}
+
+		// Now do the more expensive test initialization.
+		l.config, l.driverCleanup = driver.PrepareTest(f)
+		l.intreeOps, l.migratedOps = getMigrationVolumeOpCounts(f.ClientSet, dInfo.InTreePluginName)
+		l.resources = []*VolumeResource{}
+		l.pods = []*v1.Pod{}
+		l.stopChs = []chan struct{}{}
+
+		return l
+	}
+
+	cleanup := func(l stressTest) {
+		var errs []error
+
+		framework.Logf("Stopping and waiting for all test routines to finish")
+		for _, stopCh := range l.stopChs {
+			close(stopCh)
+		}
+		l.wg.Wait()
+
+		for _, pod := range l.pods {
+			framework.Logf("Deleting pod %v", pod.Name)
+			err := e2epod.DeletePodWithWait(cs, pod)
+			errs = append(errs, err)
+		}
+
+		for _, resource := range l.resources {
+			errs = append(errs, resource.CleanupResource())
+		}
+
+		errs = append(errs, tryFunc(l.driverCleanup))
+		framework.ExpectNoError(errors.NewAggregate(errs), "while cleaning up resource")
+		validateMigrationVolumeOpCounts(f.ClientSet, dInfo.InTreePluginName, l.intreeOps, l.migratedOps)
+	}
+
+	ginkgo.It("multiple pods should access different volumes repeatedly [Slow] [Serial]", func() {
+		const (
+			numPods = 10
```

Comment:
consider exposing these as driver config options

Reply:
If it's ok with you, I want to address this later. For now the test has to be opt-in.